src/SFA.DAS.DownloadService.UnitTests/Mappers/AparMapperRoatpResultTests.cs

```diff
@@ -0,0 +1,160 @@
+        {
+            // Arrange
+            var roatpResult = new RoatpResult { Ukprn = "invalid", OrganisationName = "Org1", ApplicationType = "main provider", StartDate = DateTime.Now };
+            Func<long, string> uriResolver = ukprn => $"http://example.com/{ukprn}";
+
+            // Act
+            var result = _mapper.Map(roatpResult, uriResolver);
+
+            // Assert
+            Assert.IsNull(result);
+        }
+
+        [Test]
+        public void Map_RoatpResultToAparEntry_EndDateBeforeToday_ReturnsNull()
+        {
+            // Arrange
+            var roatpResult = new RoatpResult { Ukprn = "12345678", OrganisationName = "Org1", ApplicationType = "main provider", EndDate = DateTime.Today.AddDays(-1), StartDate = DateTime.Now };
+            Func<long, string> uriResolver = ukprn => $"http://example.com/{ukprn}";
+
+            // Act
+            var result = _mapper.Map(roatpResult, uriResolver);
+
+            // Assert
+            Assert.IsNull(result);
+        }
+
+        [Test]
+        public void Map_ListOfRoatpResultToAparEntry_ValidInput_MapsCorrectly()
+        {
+            // Arrange
+            var startDate = DateTime.Now;
+            var applicationDeterminedDate = DateTime.Now.AddDays(-1);
+
+            var roatpResults = new List<RoatpResult>
+            {
+                new RoatpResult
+                {
+                    Ukprn = "12345678",
+                    OrganisationName = "Org1",
+                    ApplicationType = "main provider",
+                    StartDate = startDate.AddMonths(-1),
+                    ApplicationDeterminedDate = applicationDeterminedDate.AddMonths(-2),
+                    ProviderNotCurrentlyStartingNewApprentices = null,
+                },
+                new RoatpResult
+                {
+                    Ukprn = "23456789",
+                    OrganisationName = "Org2",
+                    ApplicationType = "employer provider",
+                    StartDate = startDate.AddMonths(-2),
+                    ApplicationDeterminedDate = applicationDeterminedDate.AddMonths(-3),
+                    ProviderNotCurrentlyStartingNewApprentices = DateTime.Now.AddMonths(10),
+                },
+                new RoatpResult
+                {
+                    Ukprn = "34567890",
+                    OrganisationName = "Org3",
+                    ApplicationType = "supporting provider",
+                    StartDate = startDate.AddMonths(-3),
+                    ApplicationDeterminedDate = applicationDeterminedDate.AddMonths(-4),
+                    ProviderNotCurrentlyStartingNewApprentices = null,
+                }
+            };
+            
+            Func<long, string> uriResolver = ukprn => $"http://example.com/{ukprn}";
+
+            // Act
+            var results = _mapper.Map(roatpResults, uriResolver);
+
+            // Assert
+            Assert.Multiple(() =>
+            {
+                Assert.NotNull(results);
```

Comment:
This is repeated twice, maybe you meant `Assert.NotNull(results[0])`, and then down below
`Assert.NotNull(results[1])`
`Assert.NotNull(results[2])`?

Reply:
Fixed that